### PR TITLE
Added TextureManager and some smaller things.

### PR DESCRIPTION
**Pull request by Firstbober**
- Better Object.
- Invalid texture.
- Object vector.
- CMake globbing.
- Builder.
- Repaired pthreads bug (#2).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 build
-.vscode
\ No newline at end of file
+.vscode
+bin
```
Comment (devildefu) [inline .gitignore]: More unnecessary folders for .gitignore...

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 set(source ${CMAKE_SOURCE_DIR}/src)
-add_executable(pb_battleroyale 
-${source}/main.cpp
-${source}/app.cpp
-${source}/spdlog.cpp
-${source}/handlers/mainhandler.cpp)
-target_link_libraries(pb_battleroyale sfml-graphics sfml-window sfml-system)
\ No newline at end of file
+
+file(GLOB files
```
Comment (devildefu) [inline CMakeLists.txt]: globbing is really bad

File: include/object.hpp [hunk 3]
```diff
@@ -33,11 +34,18 @@ class Object {
...
+	virtual void draw(sf::RenderWindow& window) {
+		for (auto&& child : this->childs)
+			child.draw(window);
+	}
 protected:
 	sf::Vector2f position;
 	sf::Vector2f velocity;
 	sf::Sprite sprite;
 	sf::Texture texture;
+	std::vector<Object> childs;
```
Comment (devildefu) [inline include/object.hpp]: Why do we need childs?

File: include/texturemanager.hpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class TextureManager {
+private:
+    TextureManager() { SPDLOG_INFO("Initialized TextureManager"); }
+    TextureManager( const TextureManager & ) {}
+    ~TextureManager();
+
+    std::map<std::string, sf::Texture*> textures;
+public:
+    static TextureManager& i() {
```
Comment (devildefu) [inline include/texturemanager.hpp]: There should be an instance, not an i

File: src/texturemanager.cpp [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+TextureManager::~TextureManager() {
+	SPDLOG_INFO("Cleaning textures");
+	for (auto &&[key, val] : this->textures) {
+		delete val;
+	}
+}
+
+sf::Texture& TextureManager::get(std::string texture) {
+	SPDLOG_INFO("Requested texture \"{}\"", texture);
+	if (this->textures.find(texture) == this->textures.end()) {
```
Comment (devildefu) [inline src/texturemanager.cpp]: no space between `if` and `(`

Comment (Firstbober) [reply src/texturemanager.cpp]: Repaired.

File: include/texturemanager.hpp (hunk 4 above)
Comment (Firstbober) [reply include/texturemanager.hpp]: Changed.

File: CMakeLists.txt (hunk 2 above)
Comment (Firstbober) [reply CMakeLists.txt]: Removed.

**Pull request merged**